i18n(de): Add **some** tutorial pages and correct existing pages (#12497)
withastro/docs: src/content/docs/de/tutorial/6-islands/4.mdx

---
type: tutorial
title: 'Optional: Erstelle eine Content-Collection'
description: |-
  Tutorial: Baue dein erstes Astro-Blog —
  Wandle dein Blog von dateibasierter Routenführung in Content-Collections um
i18nReady: true
head:
  - tag: title
    content: 'Blog-Tutorial: Content-Collection erstellen | Doku'
---

import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro';
import Box from '~/components/tutorial/Box.astro';
import Checklist from '~/components/Checklist.astro';
import MultipleChoice from '~/components/tutorial/MultipleChoice.astro';
import PreCheck from '~/components/tutorial/PreCheck.astro';
import Option from '~/components/tutorial/Option.astro';
import { Steps } from '@astrojs/starlight/components';

> Nun, da du ein Blog mit Astros [integrierter dateibasierter Routenführung](/de/guides/routing/#statische-routen) hast, wirst du es so aktualisieren, dass es eine [Content-Collection (Inhaltssammlung)](/de/guides/content-collections/) verwendet. Content-Collections sind eine leistungsstarke Möglichkeit, Gruppen ähnlicher Inhalte – etwa Blogbeiträge – zu verwalten.

Reviewer: ```suggestion
Nun, da du einen Blog mit Astros [integrierter dateibasierter Routenführung](/de/guides/routing/#statische-routen) hast, wirst du ihn so aktualisieren, dass es eine [Content-Collection (Inhaltssammlung)](/de/guides/content-collections/) verwendet. Content-Collections sind eine leistungsstarke Möglichkeit, Gruppen ähnlicher Inhalte – etwa Blogbeiträge – zu verwalten.
```

<PreCheck>
  - Verschiebe deinen Ordner mit Blogbeiträgen nach `src/blog/`
  - Erstelle ein Schema, um das Frontmatter deiner Blogbeiträge zu definieren
  - Verwende `getCollection()`, um Inhalte und Metadaten der Blogbeiträge abzurufen
</PreCheck>

## Lernen: Astro-Seiten vs. Content-Collections

Auch wenn du Content-Collections verwendest, nutzt du weiterhin den Ordner `src/pages/` für einzelne Seiten, etwa deine „Über mich“-Seite. Wenn du jedoch deine Blogbeiträge aus diesem speziellen Ordner herausnimmst, kannst du leistungsfähigere APIs nutzen, um dein Blog-Index und einzelne Beiträge zu erzeugen und anzuzeigen.

> Gleichzeitig erhältst du in deinem Code-Editor bessere Vorschläge und Autovervollständigung, da du ein **[Schema](/de/guides/content-collections/#definieren-des-schemas-der-sammlung)** hast, das eine gemeinsame Struktur für jeden Beitrag definiert. Astro hilft dir, diese Struktur mithilfe von [Zod](https://zod.dev/), einer Schema- und Validierungs­bibliothek für TypeScript, durchzusetzen.  

Reviewer 2: ```suggestion
Gleichzeitig erhältst du in deinem Code-Editor bessere Vorschläge und Autovervollständigung, da du ein **[Schema](/de/guides/content-collections/#defining-the-collection-schema)** hast, das eine gemeinsame Struktur für jeden Beitrag definiert. Astro hilft dir, diese Struktur mithilfe von [Zod](https://zod.dev/), einer Schema- und Validierungs­bibliothek für TypeScript, durchzusetzen.  
```

        um mich schlecht fühlen zu lassen
      </Option>
      <Option isCorrect>
        um meine Sammlungen zu verstehen und zu validieren und um Editor-Werkzeuge bereitzustellen
      </Option>
      <Option>
        nur, wenn ich die `strictest`-Konfiguration in `tsconfig.json` aktiviert habe
      </Option>
    </MultipleChoice>

</Box>

Die folgenden Schritte zeigen dir, wie du das Endergebnis des Blog-Tutorials erweiterst, indem du eine Content-Collection für deine Blogbeiträge erstellst.

## Abhängigkeiten aktualisieren

Aktualisiere Astro und alle Integrationen auf die neuesten Versionen, indem du im Terminal die folgenden Befehle ausführst:

<PackageManagerTabs>
  <Fragment slot="npm">
  ```shell
  # Astro und offizielle Integrationen gemeinsam aktualisieren
  npx @astrojs/upgrade
  ```
  </Fragment>
  <Fragment slot="pnpm">
  ```shell
  # Astro und offizielle Integrationen gemeinsam aktualisieren
  pnpm dlx @astrojs/upgrade
  ```
  </Fragment>
  <Fragment slot="yarn">
  ```shell
  # Astro und offizielle Integrationen gemeinsam aktualisieren
  yarn dlx @astrojs/upgrade
  ```
  </Fragment>
</PackageManagerTabs>

## Eine Sammlung für deine Beiträge erstellen

<Steps>

> 1. Erstelle eine neue **Sammlung** (einen Ordner) namens `src/blog/`. 

Reviewer: Unsure about this one for now since we were discussing translating collections on the Discord, just keeping this here for now
```suggestion
1. Erstelle eine neue **Collection** (einen Ordner) namens `src/blog/`. 
```

Author: I translated it with Inhaltssammlung before we were discussing about this translation. So I forgot to do it here, as I just used find and replace to search for “Inhaltssammlung” instead of “Sammlung” as well.

2. Verschiebe alle bestehenden Blogbeiträge (`.md`-Dateien) von `src/pages/posts/` in diesen neuen Ordner.

> 3. Erstelle eine Datei `src/content.config.ts`, um ein [Schema zu definieren](/de/guides/content-collections/#definieren-des-schemas-der-sammlung) für deine `postsCollection`. Für den bestehenden Code aus dem Blog-Tutorial füge Folgendes in die Datei ein, um alle im Frontmatter verwendeten Eigenschaften zu definieren:

Reviewer 2: ```suggestion
3. Erstelle eine Datei `src/content.config.ts`, um ein [Schema zu definieren](/de/guides/content-collections/#defining-the-collection-schema) für deine `postsCollection`. Für den bestehenden Code aus dem Blog-Tutorial füge Folgendes in die Datei ein, um alle im Frontmatter verwendeten Eigenschaften zu definieren:
```

    ```ts title="src/content.config.ts"
    import { glob } from "astro/loaders";
    import { z, defineCollection } from "astro:content";

    const blog = defineCollection({
        loader: glob({ pattern: '**/[^_]*.md', base: "./src/blog" }),
        schema: z.object({
          title: z.string(),
          pubDate: z.date(),
          description: z.string(),
          author: z.string(),
          image: z.object({
            url: z.string(),
            alt: z.string()
          }),
          tags: z.array(z.string())
        })
    });

    export const collections = { blog };
    ```

4. Damit Astro dein Schema erkennt, beende (<kbd>Strg + C</kbd>/<kbd>Control + C</kbd>) und starte den Entwicklungs­server neu. Dadurch wird das Modul `astro:content` bereitgestellt.
</Steps>

## Seiten aus einer Sammlung erzeugen

<Steps>
1. Erstelle eine neue Seiten­datei `src/pages/posts/[...slug].astro`.  
   Markdown- und MDX-Dateien werden innerhalb einer Sammlung nicht mehr automatisch zu Seiten. Du musst daher eine Seite erstellen, die für die Ausgabe jedes einzelnen Blogbeitrags verantwortlich ist.

> 2. Füge den folgenden Code ein, um deine Sammlung [abzufragen](/de/guides/content-collections/#sammlungen-abfragen) und für jede generierte Seite die `slug` und den Seiteninhalt bereitzustellen:

Reviewer 2: ```suggestion
2. Füge den folgenden Code ein, um deine Sammlung [abzufragen](/de/guides/content-collections/#querying-collections) und für jede generierte Seite die `slug` und den Seiteninhalt bereitzustellen:
```
